tests/puzzle/test_provider.py

```diff
@@ -116,16 +116,16 @@ def test_search_for_vcs_read_setup_with_extras(provider, mocker):
     }
 
 
-def test_search_for_vcs_read_setup_raises_error_if_no_version(provider, mocker):
+def test_search_for_vcs_read_setup_with_dynamic_version(provider, mocker):
```

Comment:
Should we consider retaining the old case as  well since it is an expected failure?